### ISSUE TITLE
✅ 🐞 Off-Canvas causes Sections and Containers to add additional gap between widgets on the frontend - [ED-15091]

### ISSUE THREAD

**Issue by reedock** [elementor/elementor#27706]
### Prerequisites

- [X] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [X] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

**Case #TSD2H4G5KWBK - Off-canvas widget is showing in front end and create a gap** 

When adding an off-canvas widget using the editor a gap in the front end is created. Trying to hide the container that contain this widget in responsive mode will just bug the entire page after triggering the off-canvas widget in the front end.

This off-canvas widget **shouldn't** create a space in the front end. 

### Screenshots
Editor:
https://snipboard.io/ERvnX9.jpg
Front end with off-canvas widget:
https://snipboard.io/tjQvAd.jpg
Front end Without off-canvas widget:
https://snipboard.io/uxDj8U.jpg

### Steps to reproduce

1. Add the off-canvas widget in the editor between 2 items (headers/text)
2. Look in front end a gap has been created..

### Expected behavior

Off-canvas widget shouldn't be taking space between other elements in the front end.

### Isolating the problem

- [ ] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [X] This bug happens with the Hello Elementor theme active.
- [X] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

<details>
<summary>Click to reveal</summary>

````txt

== Server Environment ==
	Operating System: Linux
	Software: LiteSpeed
	MySQL version: MariaDB Server v10.6.18
	PHP Version: 8.0.30
	PHP Memory Limit: 1024M
	PHP Max Input Vars: 2000
	PHP Max Post Size: 1024M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.5.4
	WP Multisite: No
	Max Upload Size: 1 GB
	Memory limit: 40M
	Max Memory limit: 1024M
	Permalink Structure: /%postname%/
	Language: en_US
	Timezone: Asia/Manila
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 3.0.2
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en-US
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36

== Active Plugins ==
	Code Snippets
		Version: 3.6.5.1
		Author: Code Snippets Pro

	Elementor
		Version: 3.22.1
		Author: Elementor.com

	Elementor Pro
		Version: 3.22.0
		Author: Elementor.com

	LiteSpeed Cache
		Version: 6.2.0.1
		Author: LiteSpeed Technologies

	Rank Math SEO with AI SEO Tools
		Version: 1.0.221
		Author: Rank Math

	Wordfence Security
		Version: 7.11.6
		Author: Wordfence


== Features ==
	Custom Fonts: 1
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Improved Asset Loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Active by default
	Additional Custom Breakpoints: Active by default
	Flexbox Container: Active by default
	Grid Container: Active by default
	Upgrade Swiper Library: Active by default
	Nested Elements Performance: Inactive by default
	Optimized Control Loading: Active
	Optimized Markup: Inactive by default
	Plugin Onboarding: Active by default
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Active by default
	Elementor Home Screen: Active by default
	Editor Top Bar: Inactive by default
	Build with AI: Inactive
	Landing Pages: Active by default
	Nested Elements: Active
	Lazy Load Background Images: Inactive by default
	Element Caching: Inactive by default
	Conversion Center: Inactive by default
	Display Conditions: Active by default
	Form Submissions: Active by default
	Menu: Inactive by default
	Taxonomy Filter: Active by default

````

</details>

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by Flickimp** [comment]
I don't get that issue at all.
I have put the Of Canvas between the Icon and Button.
No gap.

![Screenshot 2024-06-18 at 11 41 51](https://github.com/elementor/elementor/assets/86122967/8b43b628-f560-45d0-99fa-b0a43533fecc)
![Screenshot 2024-06-18 at 11 42 11](https://github.com/elementor/elementor/assets/86122967/54cdb5db-4b87-46de-a95c-b553f53c5466)

**Comment by reedock** [comment]
Editor:
https://snipboard.io/ERvnX9.jpg
Front end with off-canvas widget:
https://snipboard.io/tjQvAd.jpg
Front end Without off-canvas widget:
https://snipboard.io/uxDj8U.jpg

**Comment by reedock** [comment]
@Flickimp Followed his instructions and this gap is created only if the off-canvas widget is not placed on a 0px container. Maybe something can be tweaked so the widget can be placed anywhere in the editor without adding a gap in the front end and so bypass the settings of the container layout?

**Comment by nicholaszein** [comment]
Hello @reedock! 

Thank you for submitting your issue! 🙏 

We are aware of this issue and our team is already working on a fix. We can’t say exactly when it'll be sorted out, but we kindly ask for your patience. Thanks for understanding! 😉

A possible workaround is to add the following custom CSS to your Elementor Editor > Site Settings > Custom CSS section:

```css
.elementor-element.elementor-widget-off-canvas {
  position: absolute;
}
```

Best regards

**Comment by nicholaszein** [comment]
PS: @efintina, regarding [your report](https://github.com/orgs/elementor/discussions/27371#discussioncomment-9756895)! 😉

**Comment by reedock** [comment]
Hi @nicholaszein and thank you for your reply.

**Comment by efintina** [comment]
Thank's a lot! It works!

**Comment by bhambrabland** [comment]
Morning, I am just adding in my two cents on this as I also found this issue when placing the widget.

This is definitely something that needs to be fixed asap. We can't simply patch every site we intend to use the off canvas widget on with a workaround longer term, but I believe you are working on a fix for this already, so that is great! :-)

**Comment by nicholaszein** [comment]
Hey guys!

We have great news! 🙌 

### 📢 We're happy to announce that this issue was solved with the release of Elementor Pro v3.25.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
